### ISSUE TITLE
Root lost after updating to the latest version.

### ISSUE THREAD

**Issue by Saulildo** [HuskyDG/magisk-files#2]
so magisk delta 26.4 is out. I downloaded the latest apk and after that I downloaded magisk using the direct method. I rebooted and after that I go into the settings app and it says that the root is lost! I go into 26.4 app and the root is gone. What do I do?

**Comment by HuskyDG** [comment]
Executing su in Terminal

**Comment by Saulildo** [comment]
"No daemon is currently running!"

**Comment by Saulildo** [comment]
[magisk_log_2023-11-24T07.09.32.log](https://github.com/HuskyDG/magisk-files/files/13457862/magisk_log_2023-11-24T07.09.32.log)
Magisk logs if necessary

**Comment by Saulildo** [comment]
![Screenshot_20231124-055343](https://github.com/HuskyDG/magisk-files/assets/104403007/c9a3a727-42db-4c23-8463-fbd5d080db5f)
That's what I got.
If I download the default boot.img from my phone and flash it with the current Magisk delta, then go to fastboot and replace it with the following commands:
fastboot flash boot_a magisk_patched.img
fastboot flash boot_b magisk_patched.img
Will this process work and resolve my problem or not?

**Comment by rushiranpise** [comment]
try clean flashing magisk delta
uninstall magisk completely or flash the roms boot image
and then flash delta

**Comment by Bennett-69** [comment]
I had a similar problem: apps previously granted root access no longer had it, but also didn't request it for me to approve. Tried executing su in Terminal, it said no su executable was found.

Suspect there's some problem with Enforce SuList, as root apps work okay again with it disabled.

**Comment by Bennett-69** [comment]
> Can you boot to recovery and claim /cache/magisk.log?

I grabbed two versions for you, one with Enforce SuList enabled, and one without. Hope they help.

[EnforceSuList-magisk.log](https://github.com/HuskyDG/magisk-files/files/13466273/EnforceSuList-magisk.log)
[NoEnforceSuList-magisk.log](https://github.com/HuskyDG/magisk-files/files/13466274/NoEnforceSuList-magisk.log)

**Comment by Bennett-69** [comment]
It looks to me like the configured Superuser list in the Manager app isn't even being processed? With Enforce SuList enabled, it only adds these three default apps, not the dozen others I have enabled in the Superuser list:

sulist add: [io.github.huskydg.magisk/io.github.huskydg.magisk]
sulist add: [com.android.systemui/com.android.systemui]
sulist add: [com.android.settings/com.android.settings]

**Comment by Bennett-69** [comment]
> [#4 (comment)](https://github.com/HuskyDG/magisk-files/issues/4#issuecomment-1826788039)

That new debug build still has the same problem for me. Hopefully its more detailed logs are more useful to you.

![Screenshot_20231126-124725](https://github.com/HuskyDG/magisk-files/assets/106891950/5616903d-b715-468c-835e-26c6318b55ef)
![Screenshot_20231126-123857](https://github.com/HuskyDG/magisk-files/assets/106891950/4d038f82-02b5-40b6-8d33-beb85e29df95)
[EnforceSuList-Debug-magisk.log](https://github.com/HuskyDG/magisk-files/files/13467637/EnforceSuList-Debug-magisk.log)
[NoEnforceSuList-Debug-magisk.log](https://github.com/HuskyDG/magisk-files/files/13467638/NoEnforceSuList-Debug-magisk.log)

**Comment by Saulildo** [comment]
> try clean flashing magisk delta uninstall magisk completely or flash the roms boot image and then flash delta

If I download the boot.img of my device's rom, patch it with the default magisk, flash it, and then download the magisk delta as a module will it also work?

**Comment by Bennett-69** [comment]
> [app-debug.apk.zip](https://github.com/HuskyDG/magisk-files/files/13467917/app-debug.apk.zip)

Sorry, same behavior after installing this latest debug build. Let me know if you need new logs again.

One thing I noticed that I didn't mention before, in case it gives you any ideas: When Enforce SuList is off, I get toast notifications when any app requesting root is granted it. When Enforce SuList is on, I don't get *any* toast notifications, not even that apps are being denied root.

**Comment by Bennett-69** [comment]
> Waste my time because you don't know how to use Sulist. Your issue is a different topic

Sorry, how so? Before installing these latest builds, only apps listed under the "Superuser" tab (the SuList?) were granted root with Enforce SuList enabled. Upon installing 26401, no apps had root anymore, and only disabling Enforce SuList restored root access to them (and all other apps, obviously).

This seems like exactly the issue being reported by others in this thread, sorry if I'm mistaken.

**Comment by Bennett-69** [comment]
> > Sorry, how so? Before installing these latest builds, only apps listed under the "Superuser" tab (the SuList?) were granted root with Enforce SuList enabled.
> 
> @Bennett-69 You literally have skill reading issue

Yeah, I have trouble reading things that aren't there, like "Configure SuList:"
![Screenshot_20231127-005444~2](https://github.com/HuskyDG/magisk-files/assets/106891950/2773e658-58ee-4c43-9d57-8675ef65e0a2)

Sorry for trying to point out that you may have missed something while completely redesigning/inverting SuList vs. MagiskHide behavior in 26401. I'm obviously not the only one for whom it broke root, I didn't even open this issue myself.

**Comment by Bennett-69** [comment]
> Should fix in latest release

Just wanted to confirm this is indeed fixed for me. Previously I never *had* a "Configure SuList" option appear, which is why I mistakenly thought the existing "Superuser" list was it.

One suggestion: Just like you currently convert existing DenyList entries to MagiskHide entries upon upgrade from official Magisk to Kitsune, you might consider also creating an initial SuList from the existing Superuser entries (since those are all already allowed root until Enforce SuList is enabled).

Thanks!
